src/arch/e2k/e2k_interrupt.c

```diff
@@ -5,8 +5,74 @@
  * @date Mar 13, 2018
  * @author Anton Bondarev
  */
+#include <assert.h>
+#include <kernel/printk.h>
+#include <kernel/irq.h>
+#include <asm/ptrace.h>
+#include <util/log.h>
 
-void cpuVecHandlerNorm(void *core)
-{
+#include <drivers/interrupt/lapic/regs.h>
 
+#define MAX_TIRS 8 /* Don't know maximum TIRS count, so define 8 */
+#define GET_NR_TIRS(tir_hi)		((tir_hi >> 56) & 0xff)
+
+#define E2K_EXP_INTERRUPT  (1UL << 32)
+
+struct e2k_tir {
+	uint64_t lo;
+	uint64_t hi;
+};
+
+static void e2k_get_TIRs(struct e2k_tir *tirs) {
+	int nr_tirs = -1;
+
+	log_debug("TIR regiters:\n");
+
+	do {
+		++nr_tirs;
+		tirs[nr_tirs].hi = E2K_GET_DSREG(tir.hi);
+		tirs[nr_tirs].lo = E2K_GET_DSREG(tir.lo);
+		log_debug("TIR%d: lo=0x%lx, hi=0x%lx\n",
+			nr_tirs, tirs[nr_tirs].lo, tirs[nr_tirs].hi);
+	} while(GET_NR_TIRS(tirs[nr_tirs].hi));
+
+	assert(nr_tirs < MAX_TIRS);
```

Comment:
I guess this `assert()` should be done before accessing `tirs[nr_tirs]`

Reply:
That's true, fixed.